static/js/stream_edit.js

```diff
@@ -234,14 +234,37 @@ export function invite_user_to_stream(user_ids, sub, success, failure) {
     });
 }
 
+function show_stream_subcription_info({
+    message,
+    add_class,
+    remove_class,
+    subscribed_users,
+    already_subscribed_users,
+    ignored_deactivated_users,
+}) {
+    const stream_subscription_info_elem = $(".stream_subscription_info").expectOne();
+    const html = render_stream_subscription_info({
+        message,
+        subscribed_users,
+        already_subscribed_users,
+        ignored_deactivated_users,
+    });
+    ui.get_content_element(stream_subscription_info_elem).html(html);
+    if (add_class) {
+        stream_subscription_info_elem.addClass(add_class);
+    }
+    if (remove_class) {
+        stream_subscription_info_elem.removeClass(remove_class);
+    }
+}
```

Comment:
Can you confirm that this commit is intended to have no functional changes other than adding support for `ignored_deactivated_users`?  If so, I think it'd be cleaner to have a first commit that just extracts the function, and then a second that adds the new feature; I found this commit really confusing when I first saw it.

And actually maybe a third in between those two that change the template structure to have an `if` statement around the `each`, though you might be able to replace that more cleanly with something like putting the `"Already subscribed users:"` strings in `p` tags rather than `br`s.

Reply:
`p` tags have their own default margin which combined with `line-height` will have spacing issues like the one addressed above. So using `br`s here looks better due reasons mentioned in https://github.com/zulip/zulip/pull/18957#discussion_r656454291.

Regarding:
> Can you confirm that this commit is intended to have no functional changes other than adding support for ignored_deactivated_users?

In this commit, I have only added a new function for displaying in the info widget so that we do not repeat the same logic of adding and removing text/classes from the `.stream_subscription_info` element. Another part is template refactoring, earlier we added unnecessary `br`s even when there was no user in the Already subscribed user/Subscribed user category and also there was no way of showing message along with added user information due to template structure so that template needed a change in its structure.
Seems like there is enough story behind this template change so as you suggested it deserve to be in its own commit.

(I'll update rest of commit structure as suggested.)